### PR TITLE
Auto set test name in integration tests

### PR DESCRIPTION
**Pull request by vitlibar**
### Changelog category:
- Not for changelog (changelog entry is not required)

Auto set test name in integration tests to make situations like https://github.com/ClickHouse/ClickHouse/pull/39320 not happen again.

### CODE REVIEW

File: tests/integration/helpers/cluster.py [hunk 1]
```diff
@@ -237,6 +237,18 @@ def enable_consistent_hash_plugin(rabbitmq_id):
...
+def extract_test_name(base_path):
+    """Extracts the name of the test based to a path to its test*.py file
+    Must be unique in each test directory (because it's used to make instances dir and to stop docker containers from previous run)
+    """
+    name = p.basename(base_path)
+    if name == "test.py":
+        name = ""
+    elif name.startswith("test_") and name.endswith(".py"):
+        name = name[len("test_"):(len(name)-len(".py"))]
+    return name
```
Comment (azat) [inline tests/integration/helpers/cluster.py:249]:
```suggestion
    name = p.basename(base_path)
    name = p.splitext(name)[0]
    name = name.removeprefix('test')
    name = name.strip('-_')
    return name
```

**Review by tavplubix — APPROVED** [review]

File: tests/integration/helpers/cluster.py [hunk 2]
```diff
@@ -237,6 +237,17 @@ def enable_consistent_hash_plugin(rabbitmq_id):
...
 
 
+def extract_test_name(base_path):
+    """Extracts the name of the test based to a path to its test*.py file
+    Must be unique in each test directory (because it's used to make instances dir and to stop docker containers from previous run)
+    """
+    name = p.basename(base_path)
+    name = p.splitext(name)[0]
+    name = name.removeprefix("test")
+    name = name.strip("-_")
```
Comment (tavplubix) [inline tests/integration/helpers/cluster.py:247]:
```suggestion
    if name == "test.py":
        name = ""
    elif name.startswith("test_") and name.endswith(".py"):
        name = name[len("test_"):(len(name)-len(".py"))]
```

**Comment by Felixoid** [comment]
Well, one by one.

> But this check failed, so now I have to pull this branch, run formatter, commit, push... And it will restart tests and we will wait for the fix even more...

Black gives very verbose [diff](https://s3.amazonaws.com/clickhouse-test-reports/39322/225a5c3c7e5b779ef31b6cc6061489409efd7ac7/style_check__actions_/black_output.txt) what should be applied, it's in the [report](https://s3.amazonaws.com/clickhouse-test-reports/39322/225a5c3c7e5b779ef31b6cc6061489409efd7ac7/style_check__actions_.html). I'd propose copy/paste, but you have very reasonable next question.

> Is it possible to make CI update python formatting automatically?

I thought it might cause some issues after formatting a few months ago, but now it looks pretty stable. Having automatic formatting commit would instead help than harm. I feel that we can try autoformatting in style-check instead of checking it.

>  I would prefer to completely disable this check. It seems to me that harm exceeds benefits.

Sorry, but here I have the opposite opinion. Python pep8 gives dozens of opinionated options to write the same code "proper", and what we had before was a mix of `'` and `"` in the same lines, mixed style of functions parameters in definitions and calling, and so on.

Why do I like `black`? It simply says, `do this; the code will look the same everywhere`. Yes, there are not many settings. It's pretty authoritative. But I hope, auto-fix commits will solve the mentioned ache!

**Comment by azat** [comment]
>Why do I like black? It simply says, do this; the code will look the same everywhere. Yes, there are not many settings. It's pretty authoritative. But I hope, auto-fix commits will solve the mentioned ache!

Sometimes `black` makes the code really unreadable, here are few examples:

- https://github.com/ClickHouse/ClickHouse/blob/5809eca8c621c9721ff4ecb36402fb217bcce113/tests/integration/test_s3_cluster/test.py#L261-L264
- https://github.com/ClickHouse/ClickHouse/blob/07210c7b550b2517f7229cd0f5cb7ecd6d5874f1/tests/clickhouse-test#L430-L434
- https://github.com/ClickHouse/ClickHouse/blob/07210c7b550b2517f7229cd0f5cb7ecd6d5874f1/tests/integration/test_backup_restore_on_cluster/test.py#L464-L475

I wish something more conservative was used (i.e. just `pylint`)

**Comment by Felixoid** [comment]
Yes, sometimes it does. I agree and accept it.

We use pylint, it's nothing about the format. We either have the common code format and agree on it, or we have anarchy in it. Two tools would help with it: black and [yapf](https://github.com/google/yapf). I am not gatekeeping a particular tool, everybody is free to replace one by another.

But I wouldn't discuss just disabling format checking because it's `"the reason"`. Not the best global format does better job than arguing in each python-related PR about "well, I don't like single quotes, let's use double" vs "but pep8 doesn't care about it"

**Pull request merged**